.github/workflows/reusable-dev-build.yml

```diff
@@ -25,7 +29,14 @@ jobs:
   build:
     runs-on: ${{ fromJSON(inputs.runs-on) }}
     steps:
-      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - name: Checkout branch ${{ inputs.branch-name }}
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          ref: ${{ inputs.branch-name }}
+        if: inputs.branch-name != ''
```

Comment:
Checkout a branch if `branch-name != ''`. This is used for nightly job of building release branch.